### PR TITLE
Da mafia update. Mob nicknames, new fedoras, suits! (port)

### PR DESCRIPTION
**Pull request by boodaliboo**
https://github.com/tgstation/tgstation/pull/49029 ORIGINAL PR
quote from original pr "This PR adds a new contraband crate to cargo; the Cosa Nostra starter pack.
The crate contains four complete mafia outfits, four pictures of the virgin Mary, 0-4 Italian moustaches and 10% chance of a single switchblade.

Burning a picture of the Madonna, inducts you into the mafia, letting you pick a mafia nickname that is inserted between your first and last name.

Lorenzo Robusto -> Lorenzo "The Carp" Robusto

adds three new clothing items: the beige suit, beige fedora and white fedora.

The type of outfits you get from the crates are randomized about 60% of the outfits are the black version, about 15% are white and 15% are beige."
end quote
i just want to see if theres any travis errors rn
#### Changelog

// maintainer edit: fixed changelog
:cl:  Krysonism
rscadd: Added a new cargo crate and a fun gimmick to join the mafia (not a actual antag)
imageadd: added various new sprites
/:cl:

### CODE REVIEW

**Comment by boodaliboo** [comment]
some features of the original pr may not be here like resprited outfits as i had difficulty telling what we already had and whats new

**Comment by boodaliboo** [comment]
this worked fine when i tested it personally but please tell me if it fucks up in anyway

**Review by Ktlwjec1 — CHANGES_REQUESTED** [review]
capcarapace icons are changed with this too

File: code/game/objects/items/miscellaneous.dm [hunk 1]
```diff
@@ -144,4 +144,63 @@
...
+	var/new_name
+	var/space_position = findtext(joe.real_name, " ")
+	if(space_position)//Can we find a space?
+		new_name = "[copytext(joe.real_name, 1, space_position)] \"[nickname]\" [copytext(joe.real_name, space_position)]"
+	else //Append otherwise
+		new_name = "[joe.real_name] \"[nickname]\""
+	joe.real_name = new_name
+	used_up = TRUE
+	mob_mobs += joe
+	joe.say("My soul will burn like this saint if I betray my familiy. I enter alive and I will have to get out dead.", forced = /obj/item/virgin_mary)
```
Comment (Ktlwjec1) [inline code/game/objects/items/miscellaneous.dm]: family

File: code/game/objects/items/miscellaneous.dm [hunk 2]
```diff
@@ -144,4 +144,63 @@
...
+	///Has this item been used already.
+	var/used_up = FALSE
+	///List of mobs that have already been mobbed.
+	var/static/list/mob_mobs = list()
+
+
+/obj/item/virgin_mary/attackby(obj/item/W, mob/user, params)
+	. = ..()
+	var/ignition_msg = W.ignition_effect(src, user)
+	if(!ignition_msg)
```
Comment (JamieD1) [inline code/game/objects/items/miscellaneous.dm]:
```suggestion
	if(!ignition_msg || resistance_flags & ON_FIRE)
```

Comment (JamieD1) [reply code/game/objects/items/miscellaneous.dm]: Might as well combine them

File: code/game/objects/items/miscellaneous.dm [hunk 3]
```diff
@@ -144,4 +144,63 @@
...
+		return
+	if(resistance_flags & ON_FIRE)
+		return
+	user.dropItemToGround(src)
+	user.visible_message("<span class='danger'>[user] lights [src] ablaze with [W]!</span>", "<span class='danger'>You light [src] on fire!</span>")
+	fire_act()
+	if(used_up)
```
Comment (JamieD1) [inline code/game/objects/items/miscellaneous.dm]:
```suggestion
	if(used_up || !isliving(user) || !user.mind)
```

File: code/game/objects/items/miscellaneous.dm [hunk 4]
```diff
@@ -144,4 +144,63 @@
...
+		to_chat(joe, "<span class='warning'>You have already been initiated into the mafioso life.</span>")
+		return
+
+	to_chat(joe, "<span class='notice'>As you burn the picture, a nickname comes to mind...</span>")
+	var/nickname = input(joe, "Pick a nickname", "Mafioso Nicknames") as text|null
+	if(!nickname)
+		return
+	var/new_name
+	var/space_position = findtext(joe.real_name, " ")
+	if(space_position)//Can we find a space?
```
Comment (JamieD1) [inline code/game/objects/items/miscellaneous.dm:188]:
```suggestion
    new_name = "[joe.real_name] \"[nickname]\""
	if(space_position)//Can we find a space?
	    new_name = "[copytext(joe.real_name, 1, space_position)] \"[nickname]\" [copytext(joe.real_name, space_position)]"
```

Comment (JamieD1) [reply code/game/objects/items/miscellaneous.dm:188]: Might as well get rid of the else

**Review by JamieD1 — CHANGES_REQUESTED** [review]

**Comment by JamieD1** [comment]
@boodaliboo

File: code/game/objects/items/miscellaneous.dm [hunk 5]
```diff
@@ -144,4 +144,65 @@
...
+	var/static/list/mob_mobs = list()
+
+
+/obj/item/virgin_mary/attackby(obj/item/W, mob/user, params)
+	. = ..()
+	var/ignition_msg = W.ignition_effect(src, user)
+	if(!ignition_msg || resistance_flags & ON_FIRE)
+		return
+	if(resistance_flags & ON_FIRE)
+		return
```
Comment (JamieD1) [inline code/game/objects/items/miscellaneous.dm:167]:
```suggestion
```
Duplicate

File: code/game/objects/items/miscellaneous.dm [hunk 6]
```diff
@@ -144,4 +144,65 @@
...
+	user.dropItemToGround(src)
+	user.visible_message("<span class='danger'>[user] lights [src] ablaze with [W]!</span>", "<span class='danger'>You light [src] on fire!</span>")
+	fire_act()
+	if(used_up || !isliving(user) || !user.mind)
+		return
+	if(!isliving(user) || !user.mind) //A sentient mob needs to be burning it, ya cheezit.
+		return
```
Comment (JamieD1) [inline code/game/objects/items/miscellaneous.dm:174]:
```suggestion
```
Duplicate

File: code/game/objects/items/miscellaneous.dm [hunk 7]
```diff
@@ -144,4 +144,65 @@
...
+	var/mob/living/joe = user
+
+	if(joe in mob_mobs) //Only one nickname fuckhead
+		to_chat(joe, "<span class='warning'>You have already been initiated into the mafioso life.</span>")
+		return
+
+	to_chat(joe, "<span class='notice'>As you burn the picture, a nickname comes to mind...</span>")
+	var/nickname = input(joe, "Pick a nickname", "Mafioso Nicknames") as text|null
```
Comment (JamieD1) [inline code/game/objects/items/miscellaneous.dm:182]: Sanitise the input

**Review by JamieD1 — CHANGES_REQUESTED** [review]

**Comment by boodaliboo** [comment]
god i hate forgetting about these and being to lazy to fix them

**Pull request closed without merging**